app/src/main/java/com/alphawallet/app/ui/WalletConnectActivity.java

```diff
@@ -1216,11 +1254,138 @@ public void denyWalletConnect()
+        switchChainRequestId = intent.getLongExtra(C.EXTRA_WC_REQUEST_ID, -1);
+        switchChainId = intent.getLongExtra(C.EXTRA_CHAIN_ID, -1);
+        currentSessionId = intent.getStringExtra(C.EXTRA_SESSION_ID);
+        Timber.tag(TAG).d("MSG: SWITCH CHAIN: name: %s, chainId: %s", name, switchChainId);
+
+        if (currentSessionId == null || !session.getTopic().equals(currentSessionId))
+        {
+            Timber.tag(TAG).d("Wrong session");
+            return;
+        }
+        if (switchChainId == -1 || requestId == -1)
+        {
+            Timber.tag(TAG).d("Cant find data");
+        }
+        else
+        {
+            chainAvailable = EthereumNetworkBase.getNetworkInfo(switchChainId) != null;
+            if (!chainAvailable)
+            {
+                viewModel.approveSwitchEthChain(WalletConnectActivity.this, requestId, currentSessionId, switchChainId, false, false);
+            }
+            else
+            {
+                try
+                {
+                    String message = getString(R.string.request_change_chain, EthereumNetworkBase.getShortChainName(switchChainId), String.valueOf(switchChainId));
+                    Token baseToken = viewModel.getTokenService().getTokenOrBase(switchChainId, viewModel.defaultWallet().getValue().address);
+                    NetworkInfo newNetwork = EthereumNetworkBase.getNetworkInfo(switchChainId);
+                    NetworkInfo activeNetwork = EthereumNetworkBase.getNetworkInfo(client.chainIdVal());
+
+                    if (newNetwork != null && activeNetwork != null)
+                    {
+                        if (newNetwork.hasRealValue() && !activeNetwork.hasRealValue())
+                        {
+                            message += "\n" + getString(R.string.warning_switch_to_main);
+                        }
+                        else if (!newNetwork.hasRealValue() && activeNetwork.hasRealValue())
+                        {
+                            message += "\n" + getString(R.string.warning_switching_to_test);
+                        }
+                    }
+
+                    // show action sheet
+                    switchChainDialog = new ActionSheetDialog(this, this, R.string.switch_chain_request, message, R.string.switch_and_reload,
+                            switchChainDialogCallbackId, baseToken);
+
+                    switchChainDialog.setOnDismissListener(dialog -> {
+                        viewModel.approveSwitchEthChain(WalletConnectActivity.this, switchChainRequestId, currentSessionId, switchChainId, false, chainAvailable);
+                    });
+                    switchChainDialog.setCanceledOnTouchOutside(false);
+                    switchChainDialog.show();
+                    switchChainDialog.fullExpand();
+                }
+                catch (Exception e)
+                {
+                    Timber.e(e);
+                }
+            }
+        }
+    }
```

Comment:
Can you refactor this method and split it up a little - maybe perform all the validation in a separate method. that way we can at least get rid of one layer of this long conditional tree.

Reply:
Done. separated the show dialog code to another method.